e2e/test/scenarios/dashboard/reproductions/31628-responsive-scalars.cy.spec.js

```diff
@@ -0,0 +1,524 @@
+  { width: 1440, height: 800, openSidebar: true },
+  { width: 1440, height: 800, openSidebar: false },
+];
+
+const SCALAR_QUESTION = {
+  name: "31628 Question - This is a rather lengthy question name",
+  description: "This is a rather lengthy question description",
+  query: {
+    "source-table": ORDERS_ID,
+    aggregation: [["count"]],
+  },
+  display: "scalar",
+};
+
+const SCALAR_QUESTION_CARDS = [
+  { cards: createCardsRow({ size_y: 2 }), name: "cards 2 cells high" },
+  { cards: createCardsRow({ size_y: 3 }), name: "cards 3 cells high" },
+  { cards: createCardsRow({ size_y: 4 }), name: "cards 4 cells high" },
+  CARDS_SIZE_1X,
+];
+
+const SMART_SCALAR_QUESTION = {
+  name: "31628 Question - This is a rather lengthy question name",
+  description: "This is a rather lengthy question description",
+  query: {
+    "source-table": ORDERS_ID,
+    aggregation: [["count"]],
+    breakout: [
+      [
+        "field",
+        ORDERS.CREATED_AT,
+        {
+          "base-type": "type/DateTime",
+          "temporal-unit": "month",
+        },
+      ],
+    ],
+  },
+  display: "smartscalar",
+};
+
+const SMART_SCALAR_QUESTION_CARDS = [
+  { cards: createCardsRow({ size_y: 2 }), name: "cards 2 cells high" },
+  { cards: createCardsRow({ size_y: 3 }), name: "cards 3 cells high" },
+  { cards: createCardsRow({ size_y: 4 }), name: "cards 4 cells high" },
+];
+
+describe("issue 31628", () => {
+  describe("display: scalar", () => {
+    const descendantsSelector = [
+      "[data-testid='scalar-container']",
+      "[data-testid='scalar-title']",
+      "[data-testid='scalar-description']",
+    ].join(",");
+
+    VIEWPORTS.forEach(({ width, height, openSidebar }) => {
+      SCALAR_QUESTION_CARDS.forEach(({ cards, name }) => {
+        const sidebar = openSidebar ? "sidebar open" : "sidebar closed";
+
+        describe(`${width}x${height} - ${sidebar} - ${name}`, () => {
+          beforeEach(() => {
+            restore();
+            cy.viewport(width, height);
+            cy.signInAsAdmin();
+            setupDashboardWithQuestionInCards(SCALAR_QUESTION, cards);
+
+            if (openSidebar) {
+              openNavigationSidebar();
+            }
+          });
+
+          it(`should render descendants of a 'scalar' without overflowing it (metabase#31628)`, () => {
+            assertDescendantsNotOverflowDashcards(descendantsSelector);
+          });
+        });
+      });
+    });
+
+    describe("1x2 card", () => {
+      beforeEach(() => {
+        restore();
+        cy.signInAsAdmin();
+        setupDashboardWithQuestionInCards(SCALAR_QUESTION, [
+          { size_x: 1, size_y: 2, row: 0, col: 0 },
+        ]);
+      });
+
+      it("should truncate value and show value tooltip on hover", () => {
+        const scalarContainer = cy.findByTestId("scalar-container");
+
+        scalarContainer.then($element => assertIsEllipsified($element[0]));
+        scalarContainer.realHover();
+
+        popover().within(() => {
```

Comment:
nit: you could write `popover().findByText("18,760");`

Reply:
80af793aa1d5265f5f2d698de112b1922340baa0